---
title: "Explore freeze-dried samples Alpha diversity"
author: "hanna.bensch@lnu.se"
date: "`r format(Sys.time(), '%Y-%m-%d')`"
output:
  pdf_document:
    fig_caption: yes  
    toc: yes
---


```{r setup, echo=F,  message=FALSE}
knitr::opts_chunk$set(echo=TRUE, fig.path='../figures/FD/Alpha/', cache = TRUE)
ggplot2::theme_set(ggplot2::theme_bw())
```

```{r libraries, include = FALSE}
suppressPackageStartupMessages(library(readr))
suppressPackageStartupMessages(library(dplyr, warn.conflicts = FALSE))
suppressPackageStartupMessages(library(tidyr))
suppressPackageStartupMessages(library(tidyverse))
suppressPackageStartupMessages(library(purrr))
suppressPackageStartupMessages(library(stringr))
suppressPackageStartupMessages(library(ggplot2))
suppressPackageStartupMessages(library(kfigr))
suppressPackageStartupMessages(library(knitr))
suppressPackageStartupMessages(library(vegan))
suppressPackageStartupMessages(library(forcats))
suppressPackageStartupMessages(library(ggpubr))
suppressPackageStartupMessages(library(lme4))
suppressPackageStartupMessages(library(DT))
suppressPackageStartupMessages(library(harrypotter))
suppressPackageStartupMessages(library(phyloseq))
suppressPackageStartupMessages(library(ape))
suppressPackageStartupMessages(library(microbiome))
suppressPackageStartupMessages(library(picante))
```

```{r constants}
```

```{r read-data, message=FALSE, include=FALSE}
asvs <- read_tsv("../data/asv_table_FreezedriedVsFrozen.tsv", col_types =cols(
  asv = col_character(),
  sample = col_character(),
  count = col_double(),
  relab = col_double() ))

metadata <- read_csv("../data/FDvsFrozenMetadata.csv", col_types = cols(
  .default = col_character(),
  SampleDate = col_date(format = ""),
  SampleOrder = col_number()
)) %>% suppressWarnings()
taxonomy <- read_tsv("../data/taxonomy_FreezedriedVsFrozen.tsv", col_types =  cols(
  asv = col_character(),
  kingdom = col_character(),
  phylum = col_character(),
  class = col_character(),
  order = col_character(),
  family = col_character(),
  genus = col_character(),
  species = col_character()
))

head(asvs)
# Rename Sample number
metadata  <- metadata %>% dplyr::rename(OldSampleNumber = SampleNumber, SampleNumber = NewSampleNumber )
# make a character of the sample number
metadata <- metadata %>% mutate(Asample = paste("A", sample, sep= "_"))


metadata <- metadata %>% 
  left_join( 
    asvs %>% select(sample, count) %>% 
    group_by(sample) %>% summarise(NbReads = sum(count), .groups = 'drop'))

# get palette
pal <- hp(n = 8, house = "ronweasley2")
pal <- pal[c(6,4)]

#get the phylotree
spptree <- read.tree("../data/FDvsFrozenSepptree.tog.tree") 
```

# Introduction

In this study, we want to investigate if Freeze Dried samples give the same data as samples stored at -80 freezer prior library preparation. Here, we ask if the alpha diversity differ between the two sample types.

# Methods
We took 20 replicate samples from our fecal samples stored in -80 and freeze dried them at the Kalahari Research Centre. Freeze Dried samples were then transported and stored at room temperature until 16S library preparation of fecal samples and randomized on three sequencing plates together with their replicates. On all three plates, 4 negative controls were used to remove contaminant reads from samples before further analysis.


# Results


```{r alpha-div.vegan,fig.cap='**Boxplots Number of reads per samples,**Colour by Sample type.', fig.height=4,  fig.width=8, message=FALSE, warning=FALSE, echo=FALSE }
metadata %>%
  ggplot(aes(x = Treatment, y = NbReads, fill = Treatment)) +
  geom_boxplot() +
  scale_fill_manual( values = pal) +
    theme(axis.text.x = element_text(face = "bold", size =15),
          axis.text.y = element_text( size =15),
          axis.title.y = element_text(face = "bold", size =15),
        panel.background=element_rect(fill = 'white', colour='black'),
        strip.background=element_rect(fill = 'white', colour='white'),
        strip.text = element_text(face = "bold", size=15),
        panel.grid = element_blank(),
        legend.position = "none") +
  xlab("")
```



```{r minmaxreads check, eval=FALSE, include=FALSE}
#metadata %>% group_by(Treatment) %>% summarise( meanReads = mean(NbReads), minRich =min(NbReads), maxRich= max(NbReads))
#metadata  %>% summarise( meanReads = mean(NbReads), minRich =min(NbReads), maxRich= max(NbReads))

paired_t_reads <- lmer(NbReads ~ Treatment  + (1|Plate_No) +(1|SampleNumber), data = metadata)
summary(paired_t_reads)
paired_t_reads2 <- lmer(NbReads ~ 1  + (1|Plate_No) +(1|SampleNumber), data = metadata)
summary(paired_t_reads2)
anova(paired_t_reads, paired_t_reads2)
# No difference between number of reads/sample between treatment types
```




```{r make-phyloseq-object, include = FALSE}
str(asvs)
asvsp <- asvs %>% left_join(metadata %>% select(sample, Asample)) %>%
  select(asv, count, Asample) %>%
  pivot_wider(names_from = 'Asample', values_from='count', values_fil= 0) %>%
  tibble::column_to_rownames('asv') %>% as.matrix()

taxonomyp <- taxonomy %>% tibble::column_to_rownames('asv') %>% as.matrix()
metadatap <- metadata %>% tibble::column_to_rownames('Asample')

OTU = otu_table(asvsp, taxa_are_rows = TRUE)
TAX = tax_table(taxonomyp)
samples = sample_data(metadatap)
  
pseq <- phyloseq(OTU, TAX, samples) # make pseq object
pseq <- merge_phyloseq(pseq, spptree) # merge with tree

print(pseq)
pseq@phy_tree # tree is rooted

# rarefy to smallest lib size
minlib <- metadata %>% summarise(minLib = min(NbReads)) %>% pull()
pseq.rar <- rarefy_even_depth(pseq, sample.size = minlib, rngseed= 3000) # 144 OTU removed 
print(pseq.rar) # has 1624 taxa
print(pseq )# has 1768 taxa

# calculate alpha diversities
hmp.div <- alpha(pseq.rar, index = "all") # rarefied
hmp.divUR <- alpha(pseq, index = "all") # un-rarefied

# Add the rownames as a new colum for easy integration later.
hmp.div <- hmp.div %>% rownames_to_column("Asample")
hmp.divUR <- hmp.divUR %>% rownames_to_column("Asample")

pseq.rar.asvtab <- as.data.frame(pseq.rar@otu_table)
pseq.asvtab <- as.data.frame(pseq@otu_table)

pseq.rar.tree <- pseq.rar@phy_tree
pseq.tree <- pseq@phy_tree

# calculate pd
df.pd <- pd(t(pseq.rar.asvtab), pseq.rar.tree,include.root=T)
df.pdU <- pd(t(pseq.asvtab), pseq.tree,include.root=T)

hmp.div <- hmp.div %>% left_join(df.pd %>% rownames_to_column("Asample"))
hmp.divUR <- hmp.divUR %>% left_join(df.pdU %>% rownames_to_column("Asample"))

hmp.div <- hmp.div %>% left_join(metadata)
hmp.divUR <- hmp.divUR %>% left_join(metadata)

```



```{r AlphaLibSize, fig.cap='**Correlations between Alpha diversites A) ASV richness, B) Shannon index C) Faiths PD and number of reads**', fig.height=4,  fig.width=12, message=FALSE, warning=FALSE, echo=FALSE}

ggarrange(
hmp.div %>% 
  ggplot(aes(y = observed, x= NbReads)) +
  geom_point(aes(col = Treatment)) +
  scale_colour_manual( values = pal) +
  labs(y= "ASV Richness", x ="Library Size") +
  stat_cor(),
hmp.div %>% 
  ggplot(aes(y = diversity_shannon, x= NbReads)) +
  geom_point(aes(col = Treatment))+
  scale_colour_manual( values = pal) +
  labs(y= "Shannon index", x ="Library Size") +
  stat_cor(),
hmp.div %>% 
  ggplot(aes(y = PD, x= NbReads)) +
  geom_point(aes(col = Treatment))+
  scale_colour_manual( values = pal) +
  labs(y= "Faith's PD", x ="Library Size") +
  stat_cor(),
ncol = 3, common.legend = TRUE, legend = "top")

```

```{r AlphaResid, message=FALSE, warning=FALSE, include=FALSE}
# get residuals alpha div after controlling for Library size
# Richness
m1 <- lm(observed ~ NbReads, data = hmp.div)
summary(m1)
hmp.div$residRich <- resid(m1)
m2 <- lmer(residRich~ Treatment + (1|SampleNumber) + (1|Plate_No), data = hmp.div)
summary(m2)

# Shannon - however NbReads not sign
m3 <- lm(diversity_shannon ~ NbReads, data = hmp.div)
summary(m3)
#hmp.div$residShan <- resid(m3)
#m4 <- lmer(residShan~ Treatment + (1|SampleNumber) + (1|Plate_No), data = hmp.div)
#summary(m4)

# PD
m5 <- lm(PD ~ NbReads, data = hmp.div)
summary(m5)
hmp.div$residPD <- resid(m5)
m6 <- lmer(residPD ~ Treatment + (1|SampleNumber) + (1|Plate_No), data = hmp.div)
summary(m6)
```



```{r Figure 1a, eval=FALSE, fig.cap='**Alpha diversites calculated with phyloseq+micobiome package, fig.width=12, message=FALSE, warning=FALSE, A) ASV richness, B) Shannon index C) Faiths PD. Black line between paired sampes.**', fig.height=5, include=FALSE}

hmp.div$Treatment <- factor(hmp.div$Treatment, levels=c("Frozen", "Freeze-dried"))
ggarrange(
hmp.div  %>%
          ggplot(aes(x =Treatment, col= Treatment, y=residRich,  group = SampleNumber)) +
  geom_point(aes(size = 3)) + 
  geom_line( color="black",  aes(alpha = 1)) +
  scale_colour_manual( values = pal) +
      theme(axis.text.x = element_text( size =15),
          axis.text.y = element_text( size =15),
          axis.title.y = element_text( size =15),
        panel.background=element_rect(fill = 'white', colour='black'),
        strip.background=element_rect(fill = 'white', colour='white'),
        strip.text = element_text(face = "bold", size=15),
        panel.grid = element_blank() ,
        legend.position = "none"
        ) +
  ylab(expression(atop(" ", paste("Residual ASV Richness"))))+
       xlab(""),
hmp.div  %>%
          ggplot(aes(x =Treatment, col= Treatment, y=diversity_shannon,  group = SampleNumber)) +
  geom_point(aes(size = 3)) + 
  geom_line( color="black", aes(alpha = 1)) +
  ylab( expression(atop(" ", paste("Shannon index"))))+
  xlab("") +
  scale_colour_manual( values = pal) +
      theme(axis.text.x = element_text( size =15),
          axis.text.y = element_text( size =15),
          axis.title.y = element_text( size =15),
        panel.background=element_rect(fill = 'white', colour='black'),
        strip.background=element_rect(fill = 'white', colour='white'),
        strip.text = element_text(face = "bold", size=15),
        panel.grid = element_blank(),
        legend.position = "none"
        ) +
  ylim(0,5),
hmp.div  %>%
          ggplot(aes(x =Treatment, col= Treatment, y=residPD,  group = SampleNumber)) +
  geom_point(aes(size = 3)) + 
  geom_line( color="black", aes(alpha = 1)) +
  scale_colour_manual( values = pal) +
      theme(axis.text.x = element_text(size =15),
          axis.text.y = element_text( size =15),
          axis.title.y = element_text( size =15),
        panel.background=element_rect(fill = 'white', colour='black'),
        strip.background=element_rect(fill = 'white', colour='white'),
        strip.text = element_text(face = "bold", size=15),
        panel.grid = element_blank(),
        legend.position = "none"
        ) +
  ylab(expression(atop(" ", paste("Residual Faith's PD")))) +
  xlab(""),
labels = c("A", "B", "C"), nrow= 1, ncol =3)
```

```{r minmaxalpha, echo=FALSE, message=FALSE, warning=FALSE}
hmp.div %>% group_by(Treatment) %>% summarise( meanRich = mean(observed), minRich =min(observed), maxRich= max(observed),
                                               meanShan = mean(diversity_shannon), minShan =min(diversity_shannon), maxShan = max(diversity_shannon),
                                               meanPD = mean(PD), minPD =min(PD), maxPD = max(PD)) #%>% write.table("tables/AdivFDvsFrozen.csv")

```

```{r Figure 1, fig.cap='**Alpha diversites calculated with phyloseq+micobiome package, A+D) ASV richness, B+E) Shannon index C+F) Faiths PD. Black line between paired sampes in A-C, and D-F) correlation between sample pairs. ASV richness and PD show residual value after correcting for effect of library size.**', fig.height=9,  fig.width=12, message=FALSE, warning=FALSE, echo=FALSE}
# plot
hmp.div$Treatment <- factor(hmp.div$Treatment, levels=c("Frozen", "Freeze-dried"))
Fig1 <- ggarrange(
hmp.div  %>%
          ggplot(aes(x =Treatment, col= Treatment, y=residRich,  group = SampleNumber)) +
  geom_point(aes(size = 3)) + 
  geom_line( color="black",  aes(alpha = 1)) +
  scale_colour_manual( values = pal) +
      theme(axis.text.x = element_text( size =15),
          axis.text.y = element_text( size =15),
          axis.title.y = element_text( size =15),
        panel.background=element_rect(fill = 'white', colour='black'),
        strip.background=element_rect(fill = 'white', colour='white'),
        strip.text = element_text(face = "bold", size=15),
        panel.grid = element_blank() ,
        legend.position = "none"
        ) +
  ylab(expression(atop(" ", paste("Residual ASV Richness"))))+
       xlab(""),
hmp.div  %>%
          ggplot(aes(x =Treatment, col= Treatment, y=diversity_shannon,  group = SampleNumber)) +
  geom_point(aes(size = 3)) + 
  geom_line( color="black", aes(alpha = 1)) +
  ylab( expression(atop(" ", paste("Shannon index"))))+
  xlab("") +
  scale_colour_manual( values = pal) +
      theme(axis.text.x = element_text( size =15),
          axis.text.y = element_text( size =15),
          axis.title.y = element_text( size =15),
        panel.background=element_rect(fill = 'white', colour='black'),
        strip.background=element_rect(fill = 'white', colour='white'),
        strip.text = element_text(face = "bold", size=15),
        panel.grid = element_blank(),
        legend.position = "none"
        ) +
  ylim(0,5),
hmp.div  %>%
          ggplot(aes(x =Treatment, col= Treatment, y=residPD,  group = SampleNumber)) +
  geom_point(aes(size = 3)) + 
  geom_line( color="black", aes(alpha = 1)) +
  scale_colour_manual( values = pal) +
      theme(axis.text.x = element_text(size =15),
          axis.text.y = element_text( size =15),
          axis.title.y = element_text( size =15),
        panel.background=element_rect(fill = 'white', colour='black'),
        strip.background=element_rect(fill = 'white', colour='white'),
        strip.text = element_text(face = "bold", size=15),
        panel.grid = element_blank(),
        legend.position = "none"
        ) +
  ylab(expression(atop(" ", paste("Residual Faith's PD")))) +
  xlab(""),
hmp.div %>% select(residRich, Treatment, SampleNumber) %>% 
  pivot_wider( names_from = Treatment, values_from = residRich) %>% 
  ggplot(aes(y = `Freeze-dried`, x= Frozen)) +
  geom_point() +
  stat_cor(size = 5) + 
  xlim(-55, 101) +
  geom_smooth(method = "lm", se = FALSE, colour = "darkblue" ) +
  ylab(expression(atop("Residual ASV Richness", paste("Freeze-dried")))) +
  xlab(expression(atop("Residual ASV Richness", paste("Frozen")))) +
  theme(axis.text = element_text( size =15),
          axis.title = element_text(face = "bold", size =15),
        panel.background=element_rect(fill = 'white', colour='black'),
        strip.background=element_rect(fill = 'white', colour='white'),
        panel.grid = element_blank() ,
        legend.position = "none"
        ),
hmp.div %>% select(diversity_shannon, Treatment, SampleNumber) %>% 
  pivot_wider( names_from = Treatment, values_from = diversity_shannon) %>% 
  ggplot(aes(y = `Freeze-dried`, x= Frozen)) +
  geom_point() +
  stat_cor(size = 5)+ 
  geom_smooth(method = "lm", se = FALSE, colour = "darkblue" ) +
  ylab(expression(atop("Shannon index", paste("Freeze-dried")))) +
  xlab(expression(atop("Shannon index", paste("Frozen")))) +
  theme(axis.text = element_text( size =15),
          axis.title = element_text(face = "bold", size =15),
        panel.background=element_rect(fill = 'white', colour='black'),
        strip.background=element_rect(fill = 'white', colour='white'),
        panel.grid = element_blank() ,
        legend.position = "none"
        ),
hmp.div %>% select(residPD, Treatment, SampleNumber) %>% 
  pivot_wider( names_from = Treatment, values_from = residPD) %>% 
  ggplot(aes(y = `Freeze-dried`, x= Frozen)) +
  geom_point() +
  stat_cor(size = 5) +
  geom_smooth(method = "lm", se = FALSE, colour = "darkblue" ) +
  ylab(expression(atop("Residual Faith's PD", paste("Freeze-dried")))) +
  xlab(expression(atop("Residual Faith's PD", paste("Frozen")))) +
 theme(axis.text = element_text( size =15),
        axis.title = element_text(face = "bold", size =15),
        panel.background=element_rect(fill = 'white', colour='black'),
        strip.background=element_rect(fill = 'white', colour='white'),
        panel.grid = element_blank() ,
        legend.position = "none"
        ) ,
labels = c("A", "B", "C", "D", "E", "F"), nrow= 2, ncol =3, align = "v") 

Fig1
```

```{r Figure SX, fig.cap='**Excluding the outier in Shannon figure E.**', fig.height=4,  fig.width=4, message=FALSE, warning=FALSE, echo=FALSE}
# what about Shannon when excluding the outier
hmp.div %>% select(diversity_shannon, Treatment, SampleNumber) %>% 
  pivot_wider( names_from = Treatment, values_from = diversity_shannon) %>% 
  arrange(Frozen) %>% filter(SampleNumber != 4) %>%
  ggplot(aes(y = `Freeze-dried`, x= Frozen)) +
  geom_point() +
  stat_cor() +
  ylab(expression(atop("Shannon index", paste("Freeze-dried")))) +
  xlab(expression(atop("Shannon index", paste("Frozen")))) +
  geom_smooth(method = "lm", se = FALSE, colour = "darkblue" ) +
  theme(axis.text = element_text( size =15),
          axis.title = element_text(face = "bold", size =15),
        panel.background=element_rect(fill = 'white', colour='black'),
        strip.background=element_rect(fill = 'white', colour='white'),
        strip.text = element_text(face = "bold", size=15),
        panel.grid = element_blank() ,
        legend.position = "none"
        ) 
```

```{r lmer, eval=FALSE, include=FALSE}
#ASV richnees
paired_t_Nasv <- lmer(observed ~ Treatment  + scale(NbReads) + (1|Plate_No) +(1|SampleNumber), data = hmp.div)
summary(paired_t_Nasv)  # no effect treatment but library size
paired_t_Nasv2 <- lmer(observed ~ 1  + scale(NbReads)+ (1|Plate_No) +(1|SampleNumber), data = hmp.div)
summary(paired_t_Nasv2)
anova(paired_t_Nasv, paired_t_Nasv2) # no significant effect treatment

# ASV resid
paired_t_ResidNasv <- lmer(residRich ~ Treatment   + (1|Plate_No) +(1|SampleNumber), data = hmp.div)
summary(paired_t_ResidNasv) # 
paired_t_ResidNasv2 <- lmer(residRich ~ 1 + (1|Plate_No) +(1|SampleNumber), data = hmp.div)
summary(paired_t_ResidNasv2)
anova(paired_t_ResidNasv, paired_t_ResidNasv2) # no effect treatment

#Shannon including lib size
paired_t_shann <- lmer(diversity_shannon ~ Treatment + scale(NbReads) + (1|Plate_No)  + (1|SampleNumber), data =hmp.div)
summary(paired_t_shann) # no effect Nb reads
paired_t_shann2 <- lmer(diversity_shannon ~ 1 + scale(NbReads) + (1|Plate_No)  + (1|SampleNumber), data = hmp.div)
summary(paired_t_shann2)
anova(paired_t_shann, paired_t_shann2) # no effect treatment
# shannon exluding lib size (NS)
paired_t_shann <- lmer(diversity_shannon ~ Treatment  + (1|Plate_No)  + (1|SampleNumber), data =hmp.div)
summary(paired_t_shann)
paired_t_shann2 <- lmer(diversity_shannon ~ 1  + (1|Plate_No)  + (1|SampleNumber), data = hmp.div)
summary(paired_t_shann2)
anova(paired_t_shann, paired_t_shann2) # no effect treatment

# PD
paired_t_PD <- lmer(PD ~ Treatment + scale(NbReads) + (1|Plate_No) +(1|SampleNumber), data = hmp.div )
summary(paired_t_PD)  # no effect treatment but library size
paired_t_PD2 <- lmer(PD ~ 1 + scale(NbReads) + (1|Plate_No) +(1|SampleNumber), data = hmp.div )
summary(paired_t_PD2)
anova(paired_t_PD, paired_t_PD2) # no effect treatment but 

# PD resid
paired_t_residPD <- lmer(residPD ~ Treatment  + (1|Plate_No) +(1|SampleNumber), data = hmp.div )
summary(paired_t_residPD)
paired_t_residPD2 <- lmer(residPD ~ 1  + (1|Plate_No) +(1|SampleNumber), data = hmp.div )
summary(paired_t_residPD2)
anova(paired_t_residPD, paired_t_residPD2) # no effect

#No effect treatment
```





```{r Figure S4, echo=FALSE, fig.cap='**Rarefaction curves.**', fig.height=6, fig.width=10, message=FALSE, warning=FALSE}

asv_table <- asvs %>% dplyr::select(sample, asv, count) %>%
  pivot_wider(names_from = 'sample', values_from = 'count', values_fill = 0) %>%
  inner_join(taxonomy, by = 'asv') 

names(asv_table)[1] <- "OTU"

asv_table <- asv_table %>% dplyr::rename(Kingdom = kingdom, Phylum = phylum, Class = class, Order = order,  Family = family, Genus = genus, Species = species)

metadata <- metadata %>% relocate(sample) # reorder to get this first
amp_data <- ampvis2::amp_load(asv_table, metadata = metadata)
p1 <- ampvis2::amp_rarecurve(amp_data, stepsize = 1000, color_by = 'Treatment')
p1 + ylab("Number of ASVS") + scale_colour_manual( values = pal)  +
  theme(axis.text.y = element_text(size=15),
        axis.text.x = element_text(size=15, angle = 0),
        axis.title = element_text(size = 20, face = "bold"),
        legend.text = element_text(size=15),
        legend.title = element_text(size=15, face = "bold"))

# It looks ok! 
```

             
                                  
