---
title: "Shared ASVs freeze-dried vs frozen samples"
author: "hanna.bensch@lnu.se"
date: "`r format(Sys.time(), '%Y-%m-%d')`"
output:
  pdf_document:
    fig_caption: yes  
    toc: yes
---


```{r setup, echo=F,  message=FALSE}
knitr::opts_chunk$set(echo=TRUE, fig.path='figures/FD/Upset/', cache = TRUE)
ggplot2::theme_set(ggplot2::theme_bw())
```

```{r libraries, include = FALSE}
suppressPackageStartupMessages(library(readr))
suppressPackageStartupMessages(library(dplyr, warn.conflicts = FALSE))
suppressPackageStartupMessages(library(tidyr))
suppressPackageStartupMessages(library(tidyverse))
suppressPackageStartupMessages(library(purrr))
suppressPackageStartupMessages(library(stringr))
suppressPackageStartupMessages(library(ggplot2))
suppressPackageStartupMessages(library(kfigr))
suppressPackageStartupMessages(library(knitr))
suppressPackageStartupMessages(library(vegan))
suppressPackageStartupMessages(library(forcats))
suppressPackageStartupMessages(library(venneuler))
suppressPackageStartupMessages(library(UpSetR))
suppressPackageStartupMessages(library(ggpubr))
```


```{r constants}
```

```{r read-data, echo= FALSE, message=FALSE}
asvs <- read_tsv("data/asv_table_FreezedriedVsFrozen.tsv", col_types =cols(
  asv = col_character(),
  sample = col_character(),
  count = col_double(),
  relab = col_double() ))

metadata <- read_csv("data/FDvsFrozenMetadata.csv", col_types = cols(
  .default = col_character(),
  SampleDate = col_date(format = ""),
  SampleOrder = col_number()
)) %>% suppressWarnings()
# Rename Sample number
metadata  <- metadata %>% dplyr::rename(OldSampleNumber = SampleNumber, SampleNumber = NewSampleNumber )
# make a character of the sample number
metadata <- metadata %>% mutate(Asample = paste("A", sample, sep= "_"))

asvs %>% group_by(asv) %>% summarise(n =n()) %>% filter(n == 1) %>% summarise(n = sum(n)) # 735

```


# Introduction

In this study, we want to investigate if Freeze Dried samples give the same data as samples stored at -80 freezer prior library preparation. We ask if relative abundance differ between the two sample types.

# Methods
We took 20 replicate samples from our fecal samples stored in -80 and freeze dried them at the Kalahari Research Centre Freeze Dried samples were then transported and stored at room temperature until 16S library preparation of fecal samples and randomized on three sequencing plates together with their replicates. On all three plates, 4 negative controls were used to remove contaminant reads from samples before further analysis.



# Results



```{r Figure S3, fig.cap='**UpSet plot all asvs as either Freeze-dried or frozen sample.** Grey bar represent ASVs only found within a single sample.', fig.height = 5, fig.width = 5, message = FALSE, warning = FALSE, echo = FALSE}
Ups <- asvs %>% select(sample, asv) %>% left_join(metadata %>% select(sample, Treatment)) %>%
  mutate(N = 1 ) %>% select (-sample) %>% unique() %>% pivot_wider(names_from = 'Treatment', values_from = 'N', values_fill = 0) %>%
  left_join(asvs %>% select(sample, asv) %>% group_by(asv) %>% summarise(N = n()))  %>%
  data.frame()

upset(Ups,  
      nsets = 2,
      order.by = "freq",
       point.size = 3.5, line.size = 1.5,
      mainbar.y.label = "ASV Intersections", sets.x.label = "ASVs per Treatment", text.scale = c(1.3, 1, 1, 1, 1.3, 1.3),
      query.legend = "bottom",
      queries = list(
  list(
    query = elements,
    params = list("N", 0, 1),
    color = "grey", 
    active = T,
    query.name = "ASV only in one sample"
  )
  )
)


# summary stats
#951/(951+417+400) # 53 % of all asvs shared
#Ups %>% filter(Freeze.dried == 1 & Frozen == 1) %>% summarise(n()) #951
#Ups %>% filter(Freeze.dried == 1 & Frozen != 1) %>% summarise(n()) #400
#Ups %>% filter(Freeze.dried != 1 & Frozen == 1) %>% summarise(n()) #417

#Ups %>% filter(Freeze.dried == 1 & Frozen == 1) %>% select(asv) %>% inner_join(asvs) %>% summarise(sum(count)) %>% pull() # 3609098
#Ups %>% filter(Freeze.dried != 1 & Frozen == 1 | Freeze.dried != 1 & Frozen == 1) %>% select(asv) %>% inner_join(asvs) %>% summarise(sum(count)) %>% pull() # 11230
#asvs %>% summarise(sum(count)) %>% pull() # 3626584 total
```
