---
title: "Freeze-dried vs Frozen samples taxonomy composition"
author: "hanna.bensch@lnu.se"
date: "`r format(Sys.time(), '%Y-%m-%d')`"
output:
  pdf_document:
    toc: yes
---


```{r setup, echo=F,  message=FALSE}
knitr::opts_chunk$set(echo=TRUE, fig.path='figures/FD/Relab/', cache = TRUE)
ggplot2::theme_set(ggplot2::theme_bw())
```

```{r libraries, include = FALSE}
suppressPackageStartupMessages(library(readr))
suppressPackageStartupMessages(library(dplyr, warn.conflicts = FALSE))
suppressPackageStartupMessages(library(tidyr))
suppressPackageStartupMessages(library(purrr))
suppressPackageStartupMessages(library(stringr))
suppressPackageStartupMessages(library(ggplot2))
suppressPackageStartupMessages(library(kfigr))
suppressPackageStartupMessages(library(knitr))
suppressPackageStartupMessages(library(tibble))
suppressPackageStartupMessages(library(vegan))
suppressPackageStartupMessages(library(forcats))
suppressPackageStartupMessages(library(ggpubr))
suppressPackageStartupMessages(library(lme4))
suppressPackageStartupMessages(library(DT))
suppressPackageStartupMessages(library(wesanderson))
suppressPackageStartupMessages(library(harrypotter))
```

```{r constants}
```

```{r read-data, message=FALSE, include=FALSE, warnings=FALSE}
asvs <- read_tsv("data/asv_table_FreezedriedVsFrozen.tsv", col_types =cols(
  asv = col_character(),
  sample = col_character(),
  count = col_double(),
  relab = col_double() ))

metadata <- read_csv("data/FDvsFrozenMetadata.csv", col_types = cols(
  .default = col_character(),
  SampleDate = col_date(format = ""),
  SampleOrder = col_number()
)) %>% suppressWarnings()
# Rename Sample number
metadata  <- metadata %>% dplyr::rename(OldSampleNumber = SampleNumber, SampleNumber = NewSampleNumber )
# make a character of the sample number
metadata <- metadata %>% mutate(Asample = paste("A", sample, sep= "_"))

taxonomy <- read_tsv("data/taxonomy_FreezedriedVsFrozen.tsv", col_types =  cols(
  asv = col_character(),
  kingdom = col_character(),
  phylum = col_character(),
  class = col_character(),
  order = col_character(),
  family = col_character(),
  genus = col_character(),
  species = col_character()
))


## how many unclassified
# phyla
asvs %>% select(asv) %>% unique() %>% inner_join(taxonomy %>% filter(!is.na(phylum))) %>% summarise(n()) %>% pull() / asvs %>% select(asv) %>% unique() %>%  summarise(n()) %>% pull()
# family
asvs %>% select(asv) %>% unique() %>% inner_join(taxonomy %>% filter(!is.na(family)))%>% summarise(n()) %>% pull() / asvs %>% select(asv) %>% unique() %>%  summarise(n()) %>% pull()
# genus
asvs %>% select(asv) %>% unique() %>% inner_join(taxonomy %>% filter(!is.na(genus))) %>% summarise(n()) %>% pull() / asvs %>% select(asv) %>% unique() %>%  summarise(n()) %>% pull()

# rename unclassified
taxonomy  <- taxonomy %>%
  mutate(
    phylum = ifelse(is.na(phylum), sprintf("%s unclassified", kingdom), phylum),
    class = ifelse(is.na(class), sprintf("%s unclassified", str_remove(phylum, "unclassified")), class),
    order = ifelse(is.na(order), sprintf("%s unclassified", str_remove(class, "unclassified")), order),
    family = ifelse(is.na(family), sprintf("%s unclassified", str_remove(order, "unclassified")), family),
    genus = ifelse(is.na(genus), sprintf("%s unclassified", str_remove(family, "unclassified")), genus),
    species = ifelse(is.na(species), sprintf("%s unclassified", str_remove(genus, "unclassified")), species))

```

# Introduction

In this study, we want to investigate if Freeze Dried samples give the same data as samples stored at -80 freezer prior library preparation. We ask if relative abundance differ between the two sample types.

# Methods
We took 20 replicate samples from our fecal samples stored in -80 and freeze dried them at the Kalahari Research Centre. Freeze Dried samples were then transported and stored at room temperature until 16S library preparation of fecal samples and randomized on three sequencing plates together with their replicates.


# Results


```{r overview, eval=FALSE, message=FALSE, include=FALSE, warnings=FALSE}
# mean min max relab
asvs %>% left_join(taxonomy) %>% group_by(sample, phylum) %>% summarise(relab = sum(relab)) %>%
  group_by(phylum) %>% summarise(mean =mean(relab), min= min(relab), max= max(relab)) %>% arrange(desc(max)) %>% datatable() %>%
  formatRound(columns = c("mean", "min", "max"), digits = 3)
# any ASV dominating any sample?
asvs %>% left_join(taxonomy) %>% left_join(metadata) %>%  group_by(SampleNumber, Treatment, asv, phylum) %>% 
  summarise(relab = sum(relab), .groups = "drop")  %>% arrange(desc(relab)) # sample 4 high abundance one ASV both Freeze-dried and Frozen
asvs %>% filter(asv %in% (asvs %>% left_join(taxonomy) %>% left_join(metadata) %>%  group_by(SampleNumber, Treatment, asv, phylum) %>% summarise(relab = sum(relab), .groups = "drop")  %>% 
  arrange(desc(relab)) %>% slice(1) %>% select(asv)%>% pull())) # prevalent in 5 other samples

# mean min ma lib size
asvs %>% group_by(sample) %>% summarise(count =sum(count)) %>% summarise(min =min(count), max=max(count), mean=mean(count))
asvs %>% left_join(metadata) %>% group_by(sample, Treatment) %>% summarise(count =sum(count)) %>% group_by(Treatment) %>% summarise(min =min(count), max=max(count), mean=mean(count))

```




```{r seqdepth-n-asvs, fig.cap='**Sequencing depth and number of unique ASVs per sample.** Colour by Sample Type.', fig.height=5, fig.width=12, message=FALSE, warnings=FALSE, echo=FALSE}
# get palette
pal <- hp(n = 8, house = "ronweasley2")
pal <- pal[c(6,4)]

asvs %>% group_by(sample) %>% summarise(n_asvs = n(), sum_count = sum(count), .groups = 'drop') %>%
  pivot_longer(2:3, names_to = 'type', values_to = 'value') %>%
  left_join(metadata %>% dplyr::select(sample, Treatment, SampleNumber, SampleOrder, Plate_No)) %>%
  mutate(SampleNumber =  fct_reorder(SampleNumber, SampleOrder)) %>%
  ggplot(aes(SampleNumber, value, colour = Treatment, shape = Plate_No)) +
  scale_colour_manual( values = pal) +
  geom_point() +
  facet_wrap(~type, scales = 'free_x') +
  coord_flip()
```
  
  


```{r MeanRelab_sample_asv, fig.width=8, fig.height = 4, fig.cap = '** Correlation relative abundance within sample by sample storage method on n ASV (A) and Family (B) level.**', message = FALSE, echo = FALSE, warning = FALSE }
ggarrange(asvs %>%
  left_join(metadata %>% select(sample, Treatment, SampleNumber)) %>%
  mutate(asv_sample = paste(asv, SampleNumber)) %>%
  select(asv_sample, relab, Treatment) %>%
  pivot_wider(names_from = 'Treatment', values_from = 'relab', values_fill = 0) %>%
  ggplot(aes(x = log(`Freeze-dried`), y =log(Frozen))) +
  geom_point() +
  xlab("Log Relative abundance Freeze-dried sample")+
  ylab("Log Relative abundance Frozen sample")+
  geom_smooth(method='lm', formula= y~x) +
  stat_cor(aes(label = ..r.label..)) +
  ggtitle("ASV Level"),
  asvs %>%
  select(asv, sample, relab) %>%
  left_join(taxonomy %>% select(asv, family)) %>%
  group_by(sample, family) %>%
  summarise(fam_relab = sum(relab), .groups = 'drop') %>% 
  left_join(metadata %>% select(sample, Treatment, SampleNumber)) %>%
  select(-sample) %>%
  pivot_wider(names_from = 'Treatment', values_from = 'fam_relab', values_fill = 0) %>%
  ggplot(aes(x = log(`Freeze-dried`), y = log(Frozen))) +
  geom_point() +
  xlab("Log Relative abundance Freeze-dried sample")+
  ylab("Log Relative abundance Frozen sample")+ 
  geom_smooth(method='lm', formula= y~x) +
  stat_cor(aes(label = ..r.label..)) +
  ggtitle("Family Level"),
          labels = c("A", "B"),
          ncol = 2)
```


```{r Figure 5, echo=FALSE, fig.cap='**Correlation relative abundance within sample by sample treatment on n ASV level.** Facet by phylum, colour by top 10 phyla.', fig.height=10, fig.width=10, warning=FALSE, message=FALSE}

top9phyla <- asvs %>% 
  inner_join(taxonomy, by = 'asv') %>%
  group_by(phylum, sample) %>% 
  summarise(relab = sum(relab)) %>% 
  summarise(meanrelab = mean(relab)) %>% 
  ungroup() %>%
  top_n(9, meanrelab) %>% arrange(desc(meanrelab)) %>% 
  mutate(Order  = as.double( 1:nrow(.)),
         topph = case_when( Order == 9 ~ "Other phyla",
                           TRUE ~ phylum))

fooBarLabels <- top9phyla %>% select("Order", "topph") %>% deframe()


asvs %>%
  left_join(metadata %>% select(sample, Treatment, SampleNumber)) %>%
  mutate(asv_sample = paste(asv, SampleNumber)) %>%
  left_join(taxonomy %>% select(asv, phylum)) %>%
  left_join(top9phyla %>% filter(Order != "9")  %>% transmute(phylum, topph = phylum), by = 'phylum') %>%
  replace_na(list('topph' = 'Other phyla')) %>%
  select(Treatment, topph, asv_sample, relab) %>%
  left_join(top9phyla %>% select(topph, Order)) %>%
  rename(phylum = topph) %>% 
  mutate(phylum =  fct_reorder(phylum, Order)) %>%
  pivot_wider(names_from = 'Treatment', values_from = 'relab', values_fill = 0) %>%
  ggplot(aes(x = log(`Freeze-dried`), y =log(Frozen))) +
  geom_point(aes( col = phylum)) +
  scale_color_manual(labels = fooBarLabels, values = wes_palette("Darjeeling1", 9, type = "continuous")) +
  xlab("Log Relative abundance Freeze-dried") +
  ylab("Log Relative abundance Frozen") +
  stat_cor(aes(label = paste(..r.label.., ..p.label..,sep = "~`,`~")), size  = 3) +
  theme(axis.text=element_text(size=15),
        axis.title=element_text(size=15,face="bold"),
        strip.text = element_text(size=8),
        legend.position = "none") +
  facet_wrap(~phylum) +
  labs(col = "Phylum")
```



```{r Corrfam, echo=FALSE, fig.cap='**Correlation relative abundance within sample by sample treatment on n ASV level** colour by top 10 phyla.', fig.height=5, fig.width=6, message=FALSE, warning=FALSE}
asvs %>%
  left_join(metadata %>% select(sample, Treatment, SampleNumber)) %>%
  mutate(asv_sample = paste(asv, SampleNumber)) %>%
  left_join(taxonomy %>% select(asv, phylum)) %>%
  left_join(top9phyla %>% filter(Order != "9") %>% transmute(phylum, topph = phylum), by = 'phylum') %>%
  replace_na(list('topph' = 'Other phyla')) %>%
  select(Treatment, topph, asv_sample, relab) %>%
  left_join(top9phyla %>% select(topph, Order)) %>%
  rename(phylum = topph) %>% 
   mutate(phylum =  fct_reorder(phylum, Order)) %>%
  pivot_wider(names_from = 'Treatment', values_from = 'relab', values_fill = 0) %>%
  ggplot(aes(x = log(`Freeze-dried`), y =log(Frozen))) +
  geom_point(aes( col = phylum)) +
  scale_color_manual(labels = fooBarLabels, values = wes_palette("Darjeeling1", 9, type = "continuous")) +
  xlab("Log Relative abundance Freeze-dried")+
  ylab("Log Relative abundance Frozen")+
  theme(axis.text=element_text(size=15),
        axis.title=element_text(size=16,face="bold")) +
  stat_cor(method="pearson", aes(label = ..r.label..), size = 4) +
  labs(col = "Phylum")
```




```{r taxonomyplots, fig.cap = '**Stacked bar taxaplot per sample top 8 relab phylum. ** Sample 4 has very high relab of Proteobacteria.',  fig.width=12, fig.height = 5, message=FALSE, echo=FALSE, warning=FALSE}
#Taxonomy-plot to look closer at relative abundances of taxa within samples:

asvs %>%
  inner_join(taxonomy, by = 'asv') %>%
  left_join(top9phyla  %>% filter(Order != "9") %>% transmute(phylum, topph = phylum), by = 'phylum') %>%
  replace_na(list('topph' = 'Other phyla')) %>%
  left_join(metadata %>% select(sample, Treatment, SampleNumber, SampleOrder)) %>%
  left_join(top9phyla %>% dplyr::select(Order, topph)) %>% 
  mutate(SampleNumber =  fct_reorder(SampleNumber, desc(SampleOrder)),
          phylum =  fct_reorder(phylum, Order)) %>%
  ggplot(aes(x = SampleNumber, y = relab, group = as.factor(Order), fill = as.factor(Order))) +
  labs(y = "Relative abundance", fill = "Phylum", x ="Sample") +
  geom_col(position = 'fill') +
  scale_fill_manual(labels = fooBarLabels, values = wes_palette("Darjeeling1", 9, type = "continuous")) +
  coord_flip() +
  facet_grid(cols = vars(Treatment),  scales="free") +
      theme(axis.text.x = element_text( size =15),
          axis.text.y = element_text( size =15),
          axis.title.y = element_text(face = "bold", size =15),
          axis.title.x = element_text(face = "bold", size =15),
        panel.background=element_rect(fill = 'white', colour='black'),
        strip.background=element_rect(fill = 'white', colour='white'),
        strip.text = element_text(face = "bold", size=15),
        panel.grid = element_blank() ) 
```




```{r Figure S1, fig.cap = '**Stacked bar taxaplot per sample families with mean relab > 1 %. **', fig.width=12, fig.height = 10, message=FALSE, echo=FALSE, warning=FALSE}
# caculate top 20 familiy
topfam <- asvs %>% 
  inner_join(taxonomy, by = 'asv') %>%
  group_by(phylum, class, order, family, sample) %>% 
  summarise(relab = sum(relab)) %>% 
  summarise(meanrelab = mean(relab)) %>% 
  ungroup() %>%
  filter( meanrelab > 0.01) %>%
  #top_n(21, meanrelab) %>% 
  arrange(desc(meanrelab)) %>% 
  select(family, family, meanrelab) %>%
  add_row(family = NA, meanrelab = 0.01)  %>%
  mutate(Order  = as.double( 1:nrow(.)),
         topph = case_when( Order == 13 ~ "family < 1 % abundance",
                           TRUE ~ family))




fooBarLabels <- topfam %>% select("Order", "topph") %>% deframe()

asvs %>%
  inner_join(taxonomy, by = 'asv') %>%
  left_join(topfam  %>% filter(Order != "17") %>% transmute(family, topph = family), by = 'family') %>%
  replace_na(list('topph' = "family < 1 % abundance")) %>%
  #left_join(metadata %>% select(sample, Treatment, SampleNumber, SampleOrder)) %>%
  group_by(topph, sample) %>%
  replace_na(list('topph' = "family < 1 % abundance")) %>%
  summarise(relab = sum(relab)) %>%
  left_join(topfam %>% dplyr::select(Order, topph, family)) %>% 
  left_join(metadata %>% select(sample, Treatment, SampleNumber, SampleOrder)) %>%
  mutate(family =  fct_reorder(family, Order),
          sample = paste( Treatment, SampleNumber),
         sample =  fct_reorder(sample, desc(SampleOrder))) %>%
  ggplot(aes(x = sample, y = relab, group = as.factor(Order), fill = as.factor(Order))) +
  labs(y = "Relative abundance", fill = "Family", x ="Sample") +
  geom_col(position = 'fill') +
  scale_fill_manual(labels = fooBarLabels, values = wes_palette("Darjeeling1", 13, type = "continuous")) +
  coord_flip() +
      theme(axis.text.x = element_text( size =15),
          axis.text.y = element_text( size =13),
          axis.title.y = element_text(face = "bold", size =15),
          axis.title.x = element_text(face = "bold", size =15),
        panel.background=element_rect(fill = 'white', colour='black'),
        strip.background=element_rect(fill = 'white', colour='white'),
        strip.text = element_text(face = "bold", size=15),
        panel.grid = element_blank(),
        legend.text = element_text(size =13),
        legend.title = element_text(size = 13, face= "bold"))
```


```{r Figure 4, fig.width=10, fig.height = 8, fig.cap = '**Box plot taxaplot per sample families with mean relab > 1 %. **', message=FALSE, echo=FALSE, warning=FALSE}
asvs %>%
  inner_join(taxonomy, by = 'asv') %>%
  left_join(topfam  %>% filter(Order != "13") %>% transmute(family, topph = family), by = 'family') %>%
  replace_na(list('topph' = "family < 1 % abundance")) %>%
  select(sample, phylum, topph, relab) %>%
  group_by(topph, sample) %>%
  summarise(relab = sum(relab)) %>%
  ungroup()  %>%
  pivot_wider(names_from = topph, values_from =  relab, values_fill = 0) %>%
  pivot_longer(2:14, names_to = 'topph', values_to = 'relab') %>%
  left_join(metadata %>% select(sample, Treatment)) %>%
  rename(family= topph ) %>%
  left_join(taxonomy %>% select(family,phylum ) %>% unique()) %>%
  replace_na(list('phylum' = "")) %>%
  left_join(top9phyla %>% dplyr::select(Order, topph) %>% rename(phylum=topph)) %>% 
  mutate( family2 = paste(phylum, family),
          family2 =  fct_reorder(family2, desc(Order))) %>%
  ggplot(aes(x = family2, y = log(relab), fill = Treatment)) +
  labs(y = "Log Relative abundance", fill = "Treatment", x ="") +
  geom_boxplot() +
  scale_fill_manual( values = pal) +
      theme(axis.text.x = element_text( size =13),
          axis.text.y = element_text( size =13),
          axis.title.x = element_text(face = "bold", size =20),
        panel.background=element_rect(fill = 'white', colour='black'),
        strip.background=element_rect(fill = 'white', colour='white'),
        strip.text = element_text(face = "bold", size=15),
        panel.grid = element_blank(),
        legend.text = element_text(size=20),
        legend.title = element_blank(),
        legend.position = "top") +
  coord_flip()
  

```


```{r diffFdFs, eval=FALSE, fig.cap='**qqplot family level relab. **', fig.height=12, fig.width=12, message=FALSE, warning=FALSE, include=FALSE}

asvs %>%
  inner_join(taxonomy, by = 'asv') %>%
  left_join(topfam  %>% filter(Order != "13") %>% transmute(family, topph = family), by = 'family') %>%
  replace_na(list('topph' = "family < 1 % abundance")) %>%
  select(sample, phylum, topph, relab) %>%
  group_by(topph, sample) %>%
  summarise(relab = sum(relab)) %>%
  ungroup()  %>%
  pivot_wider(names_from = topph, values_from =  relab, values_fill = 0) %>%
  pivot_longer(2:14, names_to = 'topph', values_to = 'relab') %>%
  left_join(metadata %>% select(sample, Treatment)) %>%
  rename(family= topph ) %>%
  left_join(taxonomy %>% select(family,phylum ) %>% unique()) %>%
  replace_na(list('phylum' = "")) %>%
  left_join(top9phyla %>% dplyr::select(Order, topph) %>% rename(phylum=topph)) %>% 
  mutate( family2 = paste(phylum, family),
          family2 =  fct_reorder(family2, desc(Order))) %>%
    ggplot(aes(sample = relab)) +
    geom_qq_line() +
   geom_qq() + 
  facet_wrap(~family2 +Treatment, scales = "free")

```




```{r difffam, eval=FALSE, include=FALSE}
fam <- asvs %>%
  inner_join(taxonomy, by = 'asv') %>%
  left_join(topfam  %>% filter(Order != "13") %>% transmute(family, topph = family), by = 'family') %>%
  replace_na(list('topph' = "family < 1 % abundance")) %>%
  select(sample, phylum, topph, relab) %>%
  group_by(topph, sample) %>%
  summarise(relab = sum(relab)) %>%
  ungroup()  %>%
  pivot_wider(names_from = topph, values_from =  relab, values_fill = 0) %>%
  pivot_longer(2:14, names_to = 'topph', values_to = 'relab') %>%
  left_join(metadata %>% select(sample, Treatment)) %>%
  rename(family= topph ) %>%
  left_join(taxonomy %>% select(family,phylum ) %>% unique()) %>%
  replace_na(list('phylum' = "")) %>%
  left_join(top9phyla %>% dplyr::select(Order, topph) %>% rename(phylum=topph)) %>% 
  mutate( family2 = paste(phylum, family),
          family2 =  fct_reorder(family2, desc(Order))) %>%
  select(Treatment, sample, family2, relab) %>% 
  rename(family = family2)

famstat <- fam  %>%
  left_join(metadata %>% select(sample, SampleNumber)) %>%
  select(-sample) %>%
  arrange(Treatment, SampleNumber) %>%
  split(.$family) %>% 
  map( ~ wilcox.test(relab ~ Treatment, paired = TRUE, data = ., ) %>%  broom::tidy()) %>%
  bind_rows(.id = "family") %>%
  arrange(p.value) %>% 
  mutate(across(is.numeric, ~ round(., 5)),
         p.adjust = p.adjust(p.value, method="bonferroni")) # Bonferroni correction for that we did tests on 17 family groups 
famstat %>% datatable()
# write table
famstat  %>%
  select(family, p.value, p.adjust) %>%
  mutate_if(is.numeric,
                     round,
                     digits = 3) %>%
  mutate(p.value = as.character(p.value),
  p.value = case_when(p.value == 0 ~ "<0.001",
                      is.na(p.value) ~ "", 
                      TRUE ~ p.value),
  p.adjust = as.character(p.adjust),
  p.adjust = case_when(p.adjust == "0" ~ "<0.001",
                       TRUE ~ p.adjust))# %>%
#  write.csv(file("wilcoxtest_family.csv"))
```



