---
title: "Freeze-dried vs Frozen samples UniFraq beta diversity"
author: "hanna.bensch@lnu.se"
date: "`r format(Sys.time(), '%Y-%m-%d')`"
output:
  pdf_document:
    fig_caption: yes  
    toc: yes
---
  
  
```{r setup, echo=F,  message=FALSE}
knitr::opts_chunk$set(echo=TRUE, fig.path='../figures/FD/UniFraq/', cache = FALSE)
ggplot2::theme_set(ggplot2::theme_bw())
```

```{r libraries, include = FALSE}
suppressPackageStartupMessages(library(readr))
suppressPackageStartupMessages(library(dplyr, warn.conflicts = FALSE))
suppressPackageStartupMessages(library(tidyr))
suppressPackageStartupMessages(library(tidyverse))
suppressPackageStartupMessages(library(purrr))
suppressPackageStartupMessages(library(stringr))
suppressPackageStartupMessages(library(ggplot2))
suppressPackageStartupMessages(library(kfigr))
suppressPackageStartupMessages(library(knitr))
suppressPackageStartupMessages(library(vegan))
suppressPackageStartupMessages(library(forcats))
suppressPackageStartupMessages(library(ggpubr))
suppressPackageStartupMessages(library(lme4))
suppressPackageStartupMessages(library(DT))
suppressPackageStartupMessages(library(ggdendro))
suppressPackageStartupMessages(library(harrypotter))
suppressPackageStartupMessages(library(ape))
suppressPackageStartupMessages(library(phyloseq))
suppressPackageStartupMessages(library(microbiome))
```

```{r constants}
```

```{r read-data, include=FALSE, meassages=FALSE}
asvs <- read_tsv("../data/asv_table_FreezedriedVsFrozen.tsv", col_types =cols(
  asv = col_character(),
  sample = col_character(),
  count = col_double(),
  relab = col_double() ))
metadata <- read_csv("../data/FDvsFrozenMetadata.csv", col_types = cols(
  .default = col_character(),
  SampleDate = col_date(format = ""),
  SampleOrder = col_number()
)) %>% suppressWarnings()
taxonomy <- read_tsv("../data/taxonomy_FreezedriedVsFrozen.tsv", col_types =  cols(
  asv = col_character(),
  kingdom = col_character(),
  phylum = col_character(),
  class = col_character(),
  order = col_character(),
  family = col_character(),
  genus = col_character(),
  species = col_character()
))

# Rename Sample number
metadata  <- metadata %>% dplyr::rename(OldSampleNumber = SampleNumber, SampleNumber = NewSampleNumber )

# make a character of the sample number
metadata <- metadata %>% mutate(Asample = paste("A", sample, sep= "_"))
asvs <- asvs %>% left_join(metadata %>% select(sample,  Asample))

# get palette
pal <- hp(n = 8, house = "ronweasley2")
pal <- pal[c(6,4)]

#get the phylotree
spptree <- read.tree("../data/FDvsFrozenSepptree.tog.tree") 

```

## Introducion

I here calculate weighted an unweighted unifrac beta diversity based on the .tre file from the SPP analysis using the GG taxonomy. 

I calculated these with ordinate() on a phyloseq object  following examples such as https://www.nicholas-ollberding.com/post/introduction-to-the-statistical-analysis-of-microbiome-data-in-r/ & https://mibwurrepo.github.io/Microbial-bioinformatics-introductory-course-Material-2018/beta-diversity-metrics.html#phylogenetic-beta-diversity-metrics



```{r make-phyloseq object, include=FALSE}
str(asvs)
asvsp <- asvs %>% select(asv, count, Asample) %>% pivot_wider(names_from = 'Asample', values_from='count', values_fil= 0) %>%
  tibble::column_to_rownames('asv') %>% as.matrix()
str(taxonomy)
taxonomyp <- taxonomy %>% tibble::column_to_rownames('asv') %>% as.matrix()
metadatap <- metadata %>% tibble::column_to_rownames('Asample')

OTU = otu_table(asvsp, taxa_are_rows = TRUE)
TAX = tax_table(taxonomyp)
samples = sample_data(metadatap)
  
pseq <- phyloseq(OTU, TAX, samples) # make pseq object
pseq <- merge_phyloseq(pseq, spptree) # merge with tree

print(pseq)
pseq@phy_tree # tree is rooted

# save pseq for permanova table script
saveRDS(pseq, "../data/pseq.rds")

# delete objects not needed
rm(taxonomyp, metadatap,asvsp, samples, TAX, OTU)

```


```{r NMDS1, message=FALSE, warning=FALSE, include=FALSE}
ps1.rel <- microbiome::transform(pseq, "compositional") # w unifrac consider the abundances of different taxa.
wUF.ordu = ordinate(ps1.rel, method="NMDS", distance="unifrac", weighted=TRUE)
uwUF.ordu = ordinate(pseq, method="NMDS", distance="unifrac", weighted=FALSE)

wUF.ordu$stress 
uwUF.ordu$stress 

data.scores <- as.data.frame(scores(wUF.ordu))  #Using the scores function from vegan to extract the site scores and convert to a data.frame
data.scores$Asample <- rownames(data.scores)  # create a column of site names, from the rownames of data.scores
data.scores <- data.scores %>% left_join(metadata %>% dplyr::select(Asample, Treatment))

NMDSw<- data.scores %>%
  left_join(metadata %>% select(Asample, SampleNumber, Plate_No)) %>%
  ggplot(aes(x=NMDS1,y=NMDS2, color= Treatment, group = SampleNumber)) +
  geom_point( size = 4) +  
  geom_line( color="black", aes(alpha = 1)) +
   scale_colour_manual( values = pal)+
    theme(axis.title=element_text(size=18,face="bold"),
    axis.text = element_text(size=14),
    panel.background=element_rect(fill='white', colour='black'),
    strip.background=element_rect(fill='white', colour='white'),
    panel.grid = element_blank(),
    legend.key=element_blank(),
    legend.position = "right",
    legend.title = element_blank(),
    legend.text = element_text( size=12, face="bold"))+
  guides(alpha =FALSE, fill =FALSE) +
  ggtitle("A) Weighted UniFrac")


data.scores <- as.data.frame(scores(uwUF.ordu))  #Using the scores function from vegan to extract the site scores and convert to a data.frame
data.scores$Asample <- rownames(data.scores)  # create a column of site names, from the rownames of data.scores
data.scores <- data.scores %>% left_join(metadata %>% dplyr::select(Asample, Treatment))

NMDSuw<- data.scores %>%
  left_join(metadata %>% select(Asample, SampleNumber, Plate_No)) %>%
  ggplot(aes(x=NMDS1,y=NMDS2, color= Treatment, group = SampleNumber)) +
  geom_point( size = 4) +  
  geom_line( color="black", aes(alpha = 1)) +
   scale_colour_manual( values = pal)+
    theme(axis.title=element_text(size=18,face="bold"),
    axis.text = element_text(size=14),
    panel.background=element_rect(fill='white', colour='black'),
    strip.background=element_rect(fill='white', colour='white'),
    panel.grid = element_blank(),
    legend.key=element_blank(),
    legend.position = "right",
    legend.title = element_blank(),
    legend.text = element_text( size=12, face="bold"))+
  guides(alpha =FALSE, fill =FALSE) +
  ggtitle("B) Unweighted UniFrac")

```


```{r Figure S2, echo=FALSE, fig.cap='**NMDS weighted & unweigted unifrac un-rarefied data. Point colour represent the sample treatment and line beteween replicates.**', fig.height=5, fig.width=12, message=FALSE, warning=FALSE}

allsamples_WuW <- ggarrange(NMDSw, NMDSuw, common.legend = TRUE)
allsamples_WuW 
```


```{r NMDS2, message=FALSE, warning=FALSE, include=FALSE}
##
wUF.ordu = ordinate(subset_samples(ps1.rel, SampleNumber != "4"), method="NMDS", distance="unifrac", weighted=TRUE) # Weighted Unifrac consider the abundances of different taxa.
uwUF.ordu = ordinate(subset_samples(pseq, SampleNumber != "4"), method="NMDS", distance="unifrac", weighted=FALSE)

wUF.ordu$stress 
uwUF.ordu$stress

data.scores <- as.data.frame(scores(wUF.ordu))  #Using the scores function from vegan to extract the site scores and convert to a data.frame
data.scores$Asample <- rownames(data.scores)  # create a column of site names, from the rownames of data.scores
data.scores <- data.scores %>% left_join(metadata %>% dplyr::select(Asample, Treatment))

NMDSw<- data.scores %>%
  left_join(metadata %>% select(Asample, SampleNumber, Plate_No)) %>%
  ggplot(aes(x=NMDS1,y=NMDS2, color= Treatment, group = SampleNumber)) +
  geom_point( size = 4) +  
  geom_line( color="black", aes(alpha = 1)) +
   scale_colour_manual( values = pal)+
    theme(axis.title=element_text(size=18,face="bold"),
    axis.text = element_text(size=14),
    panel.background=element_rect(fill='white', colour='black'),
    strip.background=element_rect(fill='white', colour='white'),
    panel.grid = element_blank(),
    legend.key=element_blank(),
    legend.position = "right",
    legend.title = element_blank(),
    legend.text = element_text( size=12, face="bold"))+
  guides(alpha =FALSE, fill =FALSE) +
  ggtitle("A) Weighted UniFrac")


data.scores <- as.data.frame(scores(uwUF.ordu))  #Using the scores function from vegan to extract the site scores and convert to a data.frame
data.scores$Asample <- rownames(data.scores)  # create a column of site names, from the rownames of data.scores
data.scores <- data.scores %>% left_join(metadata %>% dplyr::select(Asample, Treatment))

NMDSuw<- data.scores %>%
  left_join(metadata %>% select(Asample, SampleNumber, Plate_No)) %>%
  ggplot(aes(x=NMDS1,y=NMDS2, color= Treatment, group = SampleNumber)) +
  geom_point( size = 4) +  
  geom_line( color="black", aes(alpha = 1)) +
   scale_colour_manual( values = pal)+
    theme(axis.title=element_text(size=18,face="bold"),
    axis.text = element_text(size=14),
    panel.background=element_rect(fill='white', colour='black'),
    strip.background=element_rect(fill='white', colour='white'),
    panel.grid = element_blank(),
    legend.key=element_blank(),
    legend.position = "right",
    legend.title = element_blank(),
    legend.text = element_text( size=12, face="bold"))+
  guides(alpha =FALSE, fill =FALSE) +
  ggtitle("B) Unweighted UniFrac")
```

```{r Figure 3, echo=FALSE, fig.cap='**NMDS weighted & unweigted unifrac un-rarefied data and sample 4 excluded. Point colour represent the sample treatment and line beteween replicates.**', fig.height=5, fig.width=12, message=FALSE, warning=FALSE}

WuW_fourExcluded <-  ggarrange(NMDSw, NMDSuw, common.legend = TRUE)
WuW_fourExcluded
```


